### ISSUE TITLE
How to install it?

### ISSUE THREAD

**Issue by mrfrenkie** [jbree/touchbarpiano#1]
Dear Joshua. I'm just a musician and don't know how to install it. It seems you touchbar piano can send MIDI and i really need it. Thanks for your work!

**Comment by jbree** [comment]
This is unfinished software. Here's a build of it if you'd like to mess around with it. I don't have a touch bar mac to even test this on, so it's completely unsupported. Since I don't have a touchbar, I was unable to implement multi-touch: it's monophonic only.

[TouchBarPiano.app.zip](https://github.com/jbree/touchbarpiano/files/1403597/TouchBarPiano.app.zip)

**Comment by jbree** [comment]
Side note... you may need to adjust your security settings, or right click and click "Open" in order to launch it for the first time.

**Comment by mrfrenkie** [comment]
Thank you for answer! I will can test it for you. I'm going to use this in my work daily.

At first sight this does not work. I see only the octave change window.

**Comment by jbree** [comment]
One of the limitations is that it can only work when TouchBar Piano is the front-most app. Please confirm this. The app works correctly in the simulator and, without access to a TouchBar equipped Mac, I have no way of reproducing your issue :/

**Comment by mrfrenkie** [comment]
![img_1522](https://user-images.githubusercontent.com/14099347/31856270-7ff553fe-b6c5-11e7-9527-935fe52429c0.JPG)

Oh. You're right. It work only when in focus. Notice that piano in touchbar very dark and probably I did not see it last time.

Also i have no idea when we can change midi input/output. But this thing is not so important so long as the app only works when in focus.

So that yes. it works within its functionality.

**Comment by mrfrenkie** [comment]
Your application can be an excellent portable solution for all owners of mac with touchbar. No more need to keep the caps lock on. Only Logic Pro at the moment has this functionality. So your product is potentially suitable for any DAW.

I hope "front-most app" is not an unreachable limitation. I do not understand anything about software development. And in English lol

**Comment by mrfrenkie** [comment]
I understood why piano is so dark. This is for the animation to press the keys.
Perhaps inverting this solution would be nice. When the pressed key is dark

**Comment by jbree** [comment]
Here is a brighter version.

[TouchBarPiano.app.zip](https://github.com/jbree/touchbarpiano/files/1404997/TouchBarPiano.app.zip)

**Comment by mrfrenkie** [comment]
yes! it looks great now. Tell me please, do you plan to continue working with this app? Because now it's just a beautiful piano for touchbar. It has no any useful functional.

**Comment by jbree** [comment]
I’m not sure what you mean. If you open up Logic and then Touch Bar Piano, the piano is added as a new midi input. add a software instrument and the key touches will be sent to it. Works for me.

No plans for further development unless I get a touch bar mac, which isn’t happening soon.
